src/core/types/field-descriptor.ts

```diff
@@ -50,3 +50,18 @@ export const isPrimitiveDescriptor = (
 ): boolean => {
   return !isArrayDescriptor(field) && !isObjectDescriptor(field);
 };
+
+export const getArrayDescriptorChildren = <T extends Array<unknown>, Err>(
+  descriptor: ArrayFieldDescriptor<T, Err>,
+  numberOfChildren: number
+): Array<ReturnType<typeof descriptor["nth"]>> => {
+  return Array.from(Array(numberOfChildren), (_, index) =>
```

Comment:
that's cool, but just out of curiosity, will this work?
`=> range(0, numberOfChildren - 1).map(descriptor.nth)` 

I added `range` some time ago, now it would fit into array utils maybe